app/api.go

```diff
@@ -94,50 +93,39 @@ func addHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	defer file.Close()
 
-	out, err := os.Create(header.Filename)
-	if err != nil {
-		logging.Logger.Errorf("Unable to create the file for writing. Check your write access privilege.", err)
-		fmt.Fprintf(w, "Unable to create the file for writing. Check your write access privilege.", err)
-		w.WriteHeader(http.StatusInternalServerError)
-	}
-	defer out.Close()
-
-	_, err = io.Copy(out, file)
-	if err != nil {
-		logging.Logger.Errorf("Error copying file.", err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
 	// return that we have successfully uploaded our file!
 	fmt.Fprintf(w, "Successfully Uploaded File\n")
 	logging.Logger.Info("Received file : ", header.Filename)
 
 	// fetch an GPRC connection
-	connection, err := dial(logRpcServer)
+	logRpcServer := fmt.Sprintf("%s:%d",
+		viper.GetString("trillian_log_server.address"),
+		viper.GetInt("trillian_log_server.port"))
+	connection, err := dial(ctx, logRpcServer)
 	if err != nil {
 		fmt.Printf("%+v\n", err)
 	}
 
 	leafFile, err := os.Open(header.Filename)
-
+	defer leafFile.Close()
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		logging.Logger.Fatal(err)
 	}
 
-	byteLeaf, _ := ioutil.ReadAll(leafFile)
-	defer leafFile.Close()
+	byteLeaf, err := ioutil.ReadAll(file)
+	if err != nil {
+		logging.Logger.Fatal(err)
```

Comment:
we should probably swap this to something that doesn't exit the process, and instead returns an error code/keeps running. I'll handle that next.

Reply:
good catch and very much agree!

Reply:
All set now!